lib/handler/file.c

```diff
@@ -208,19 +210,22 @@ static int do_pull(h2o_generator_t *_self, h2o_req_t *req, h2o_iovec_t *buf)
     while ((rret = pread(self->file.ref->fd, buf->base, buf->len, self->file.off)) == -1 && errno == EINTR)
         ;
     if (rret <= 0) {
-        req->http1_is_persistent = 0; /* FIXME need a better interface to dispose an errored response w. content-length */
         buf->len = 0;
         self->bytesleft = 0;
+        if (rret < 0) {
```

Comment:
Thank you for making the changes to file.c.

I assume we do not need this check, since:
* we should return an error if we reached an EOF before reading the expected number of bytes (the number which has already been indicated to the client through the `content-length` header)
* `do_pull` never gets called if `bytesleft == 0` (there's a guard in `do_send_file` for the first invocation, and for succeeding calls, the function returns `H2O_SEND_STATE_FINAL` once bytesleft becomes zero)